wordpress-mobile/WordPress-Android
# Quick Start for Existing Users V2: Add check notifications task

Parent #16350 

This PR adds the "Check your notifications" task to the new `Quick Start` card.

Corresponding FluxC PR: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2395

https://user-images.githubusercontent.com/1405144/168190498-d5cece95-2378-499c-86c2-f7ab4c054bf9.mp4

https://user-images.githubusercontent.com/1405144/168191341-1f65a4a4-617d-49cf-835e-450e78180681.mp4

To test:

/cc @jostnes, @osullivanchris 

Test.1 Select "Check your notifications" task from the tasks list

1. Launch the app.
2. Enable to `Me` -> `App Settings` -> `Debug settings` -> `QuickStartForExistingUsersV2FeatureConfig`.
3. Restart the app.
4. Logout/login and select "Show me around" when the quick start prompt is shown.
5. Select an existing site.
6. Go to `My Site` tab.
7. ✅ Notice that "Check your notifications" task is shown in the new `Quick Start` card with `Get to know the WordPress app` task type.
8. Tap on `Get to know the WordPress app` in the app.
9. Tap on "Check your notifications" task. 
10. ✅ Notice that 
    - snackbar is displayed to select the notifications tab.
    - quick start focus point is displayed on the notifications tab.
11. Select `Notifications` tab.
12. ✅ Notice that quick start focus point is gone.
13. Return to `My Site` tab.
14. ✅ Notice that the complete tasks count is incremented at the bottom of the `Quick Start` card.
15. Tap on "Check your notifications" task. 
16. ✅ Notice that "Check your notifications" task has moved to the completed tasks list.

Test.2 Snackbar with "Go" button

1. Repeat steps 4-8 from `Test.1`.
2. Select "Check your site stats" task.
3. Follow the instructions to complete the "Check your site stats" task.
4. ✅ Notice that a snackbar is displayed  "Check your notifications" task with a "Go" button. 
5. Tap on the "Go" button in the snackbar.
6. Repeat steps 10-16 from `Test.1`

PS: Currently a placeholder illustration is added for the quick start dynamic card for the "Check your notifications" task. Since the dynamic card is hidden behind a feature flag, we can wait to add the actual illustration later.

## Merge Instructions

1. Make sure #16530 is merged to trunk. ✅ 
2. Make sure `FluxC` PR https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2395 is merged to trunk.  ✅ 
3. Update `FluxC` version in build.gradle.  ✅ 
4. Remove `Not Ready for Merge` label.  ✅ 
5. Merge the PR.

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A 

3. What automated tests I added (or what prevented me from doing so)
Added test for check notifications quick start focus point. More comprehensive tests will be added later.

PR submission checklist:
